### ISSUE TITLE
Allow the package update workflow to consider the set of latest version registered on the website listing if it is compatible and newer than `latest`

### ISSUE THREAD

**Issue by dev7355608** [foundryvtt/foundryvtt#8115]
### What happened?

Since I released Perfect Vision 4.0.0, which is only V10-compatible, the package updater fails to update Perfect Vision to the latest v9 version (3.9.15) for users on V9: The updater says the latest version requires V10. It doesn't seem check the package listings for the latest compatible version and instead checks only whether the latest version is compatible but stops there. It requires a reinstall of the module to get the latest V9-compatible version. The installer is able to select the correct version, but the updater is not.

![image](https://user-images.githubusercontent.com/31905376/188686163-1ec2e5b8-43c2-4950-b264-1f16703d3365.png)

https://raw.githubusercontent.com/dev7355608/perfect-vision/v3.9.14/module.json
https://raw.githubusercontent.com/dev7355608/perfect-vision/v3.9.15/module.json
https://raw.githubusercontent.com/dev7355608/perfect-vision/v4.0.0/module.json

### What ways of accessing Foundry can you encounter this issue in?

- [ ] Native App (Electron)
- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Other

### Reproduction Steps

1. Install Foundry 9.280 and PV 3.9.14.
2. Try to update PV.

Result: Blocked. But 3.9.15 is listed as the latest V9-compatible version.

### What core version are you reporting this for?

Version 10 Stable 1

### Bug Checklist

- [X] The issue occurs while all Modules are disabled

**Comment by ghost91-** [comment]
I ran into the same issue with FXMaster, where I released 2.7.1 as a bugfix for the v9 version, after having already released the v10 only 3.0.0. Users who are still in v9 get 2.7.1, when doing a fresh install of FXMaster, but users who already have 2.7.0 installed don’t get 2.7.1 suggested as an update. They just see the message about a newer version being available, but requiring foundry v10.

Now, I understand that this would be the case with the regular manifest url upgrading mechanism, since the URL just points to the v10 version. But it would be very helpful if it worked via the side grading from the package admin. It seems to work for new installations already, after all.
